### ISSUE TITLE
Tests\Feature\Api\DeleteAccountTypeTest::testDisableAccountTypeThatDoesNotExist failing intermittently 

### ISSUE THREAD

**Issue by jdenoc** [jdenoc/money-tracker#33]
This [test](https://github.com/jdenoc/money-tracker/blob/01e1a610f5a6addbde61eacdc49e8a51b64e031d/tests/Feature/Api/DeleteAccountTypeTest.php#L20-L30) has been failing intermittently on travis-ci.

Test failure output:
```
1) Tests\Feature\Api\DeleteAccountTypeTest::testDisableAccountTypeThatDoesNotExist
Expected status code 404 but received 204.
Failed asserting that false is true.
/home/travis/build/jdenoc/money-tracker/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestResponse.php:79
/home/travis/build/jdenoc/money-tracker/tests/Feature/Api/DeleteAccountTypeTest.php:29
```

**Comment by jdenoc** [comment]
Current theory is that the database isn't being "reset" before this test is run.

**Comment by jdenoc** [comment]
**Suggestion:** Find out what database tests are being run against. Maybe create an `artisan` script to do this.

**Comment by jdenoc** [comment]
Here is some output of a failed build from [travis-ci build 70](https://travis-ci.org/jdenoc/money-tracker/builds/299735474):
```
0.20s$ php artisan travis-ci --display-env
Environment Variables:
+-------------------+-----------------------------------------------------+
| variable          | value                                               |
+-------------------+-----------------------------------------------------+
| APP_ENV           | local                                               |
| APP_KEY           | base64:4SedTOQE0uj45VuzHdJ2iaBGWriPThB/JuAhgFSMG8o= |
| APP_LOG_LEVEL     | debug                                               |
| BROADCAST_DRIVER  | log                                                 |
| CACHE_DRIVER      | file                                                |
| SESSION_DRIVER    | file                                                |
| QUEUE_DRIVER      | sync                                                |
| REDIS_HOST        | 127.0.0.1                                           |
| REDIS_PASSWORD    |                                                     |
| REDIS_PORT        | 6379                                                |
| MAIL_DRIVER       | smtp                                                |
| MAIL_HOST         | mailtrap.io                                         |
| MAIL_PORT         | 2525                                                |
| MAIL_USERNAME     |                                                     |
| MAIL_PASSWORD     |                                                     |
| MAIL_ENCRYPTION   |                                                     |
| PUSHER_APP_ID     |                                                     |
| PUSHER_APP_KEY    |                                                     |
| PUSHER_APP_SECRET |                                                     |
| APP_DEBUG         |                                                     |
| APP_URL           | http://money-tracker.jdenoc.dev                     |
| DB_CONNECTION     | mysql                                               |
| DB_HOST           | localhost                                           |
| DB_PORT           | 3306                                                |
| DB_DATABASE       | money_tracker                                       |
| DB_USERNAME       | jdenoc                                              |
| DB_PASSWORD       | password                                            |
+-------------------+-----------------------------------------------------+
before_script.1
0.01s$ mysql -u root -e "SHOW DATABASES;"
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
| travis             |
+--------------------+
before_script.2
0.01s$ mysql -u root -e "CREATE DATABASE money_tracker;"
before_script.3
0.01s$ mysql -u root -e "CREATE USER 'jdenoc'@'localhost' IDENTIFIED BY 'password';"
before_script.4
0.01s$ mysql -u root -e "GRANT ALL ON money_tracker.* TO 'jdenoc'@'localhost';"
before_script.5
0.01s$ mysql -u root -e "SHOW DATABASES;"
+--------------------+
| Database           |
+--------------------+
| information_schema |
| money_tracker      |
| mysql              |
| performance_schema |
| travis             |
+--------------------+
464.48s$ travis_wait vendor/bin/phpunit --coverage-text

The command vendor/bin/phpunit --coverage-text exited with 1.

Log:
PHPUnit 5.7.23 by Sebastian Bergmann and contributors.
F..............................................................  63 / 127 ( 49%)
............................................................... 126 / 127 ( 99%)
.                                                               127 / 127 (100%)

Time: 7.73 minutes, Memory: 129.75MB
There was 1 failure:
1) Tests\Feature\Api\DeleteAccountTypeTest::testDisableAccountTypeThatDoesNotExist
Expected status code 404 but received 204.
Failed asserting that false is true.
/home/travis/build/jdenoc/money-tracker/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestResponse.php:79
/home/travis/build/jdenoc/money-tracker/tests/Feature/Api/DeleteAccountTypeTest.php:29

FAILURES!
Tests: 127, Assertions: 67199, Failures: 1.

The command "travis_wait vendor/bin/phpunit --coverage-text" exited with 1.
after_script.1
0.01s$ mysql -u root -e "SHOW DATABASES;"
+--------------------+
| Database           |
+--------------------+
| information_schema |
| money_tracker      |
| mysql              |
| performance_schema |
| travis             |
+--------------------+
after_script.2
0.01s$ mysql -u root -e "DROP USER 'jdenoc'@'localhost';"
after_script.3
0.02s$ mysql -u root -e "DROP DATABASE money_tracker;"
after_script.4
0.01s$ mysql -u root -e "SHOW DATABASES;"
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
| travis             |
+--------------------+
Done. Your build exited with 1.
```

We see that the environment variables are set correctly and database does not exist before tests are run.

**Comment by jdenoc** [comment]
Seems that the cause of this problem is due to the need to call `setUp()` within a provide method. It is called [here](https://github.com/jdenoc/money-tracker/blob/01e1a610f5a6addbde61eacdc49e8a51b64e031d/tests/Feature/Api/PostEntriesTest.php#L14), [here](https://github.com/jdenoc/money-tracker/blob/01e1a610f5a6addbde61eacdc49e8a51b64e031d/tests/Feature/Api/PostEntryTest.php#L41) and [here](https://github.com/jdenoc/money-tracker/blob/01e1a610f5a6addbde61eacdc49e8a51b64e031d/tests/Feature/Api/PutEntryTest.php#L185).

Need to figure out how to have the `setUp()` method for those test classes ONLY before the provider methods are called.

**Comment by jdenoc** [comment]
Looks like we found a solution. Instead of calling `setUp()`, We'll just initialise the application outside of the `setUp()` method.

**Comment by jdenoc** [comment]
Fixed in [0.2.0](https://github.com/jdenoc/money-tracker/releases/tag/0.2.0)
